dynamed/parser.js

```diff
@@ -21,14 +21,16 @@ module.exports = new Parser(function analyseEC(parsedUrl, ec) {
 
   let match;
 
-  if ((match = /^\/results$/i.exec(path)) !== null) {
+  if (/^\/results$/i.test(path)) {
     // https://www.dynamed.com/results?q=heart&lang=en
-    result.rtype    = 'SEARCH';
-    result.mime     = 'HTML';
-  } else if ((match = /^\/api\/BuzzDmpSearch$/i.exec(path)) !== null) {
-    // http://www.dynamed.com:80/api/BuzzDmpSearch
-    result.rtype    = 'SEARCH';
-    result.mime     = 'HTML';
+    result.rtype = 'SEARCH';
+    result.mime = 'HTML';
+  } else if (/^\/api\/search|BuzzDmpSearch(\/v6\/search)?$/i.test(path)) {
```

Comment:
it is correct but I think that to cut this parser in 2 parsers would allow to improve the reading of the code